src/battle_script_commands.c

```diff
@@ -8121,8 +8121,13 @@ static bool32 HasAttackerFaintedTarget(void)
 
 bool32 CanPoisonType(u8 battlerAttacker, u8 battlerTarget)
 {
-    return ((GetBattlerAbility(battlerAttacker) == ABILITY_CORROSION && gBattleMoves[gCurrentMove].split == SPLIT_STATUS)
-            || !(IS_BATTLER_OF_TYPE(battlerTarget, TYPE_POISON) || IS_BATTLER_OF_TYPE(battlerTarget, TYPE_STEEL)));
+    if (GetBattlerAbility(battlerAttacker) == ABILITY_CORROSION
```

Comment:
This is extraneous. You can just skip the check for returning FALSE on Poison or Steel types if the attacker's Ability is Corrosion. Something like "ability != corrosion && (type == poison || type == steel) then return false"

Reply:
Oh damn, right.

Reply:
pushed a change, thx